src/components/generic/Layout/DashboardLayout.tsx

```diff
@@ -1,20 +1,78 @@
-import { DetailedHTMLProps, HTMLAttributes, PropsWithChildren } from "react";
+import { useRouter } from "next/router";
+import {
+  cloneElement,
+  DetailedHTMLProps,
+  HTMLAttributes,
+  PropsWithChildren,
+  ReactElement,
+  useEffect,
+  useState
+} from "react";
 
+import { useGetV2DashboardCountries } from "@/generated/apiComponents";
 import HeaderDashboard from "@/pages/dashboard/components/HeaderDashboard";
 
 import Sidebar from "../Sidebar/Sidebar";
 
 interface DashboardLayoutProps extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
   isLoggedIn?: boolean;
+  user?: { name: string; email: string }; // Example data to pass
+}
+
+export interface CountriesProps {
+  data: {
+    label: string;
+    icon: string;
+  };
+  id: number;
+  country_slug: string;
 }
 
 const DashboardLayout = (props: PropsWithChildren<DashboardLayoutProps>) => {
+  const router = useRouter();
+  const { data: dashboardCountries } = useGetV2DashboardCountries<any>({
```

Comment:
you can replace `any` with a type that better reflects the structure of the data. Assuming `dashboardCountries` is an object with a `data` property that is an array of `CountriesProps`, you can type it like this:
```
const { data: dashboardCountries } = useGetV2DashboardCountries<{ data: CountriesProps[] }>({
  queryParams: {}
});
```
I'm not sure what the structure is, so it should be review.

Reply:
done